### PR TITLE
Provide detector-dependent spectrum masks

### LINKED ISSUE #19685: Spectra masks are not set on a specific detector in the ISIS SANS reduction backend
Changes in the `SpectrumNumberTranslator` have highlighted that there is an issue in the spectrum masking. The spectra are not set on a detector but in a general matter. However we need to determine to which detector a particular spectrum number is associated.

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
Fixes #19685 

#### To test

Code review. Note that part of the implementation is not very nice, but goes along the lines of [this](https://github.com/mantidproject/mantid/blob/master/scripts/SANS/isis_reduction_steps.py#L695:L711)
Make sure that the unit and system tests pass.

Does not have to be in release notes. The changes have not been released yet.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/sans/common/general_functions.py [hunk 1]
```diff
@@ -773,3 +774,24 @@ def does_can_workspace_exist_on_ads(can_workspace):
...
+    :returns: either LAB or HAB
+    """
+    detector = DetectorType.LAB
+    if instrument is SANSInstrument.LOQ:
+        if 16387 <= spectrum_number <= 17784:
+            detector = DetectorType.HAB
+    elif instrument is SANSInstrument.LOQ:
+        if 36873 <= spectrum_number <= 73736:
+            detector = DetectorType.HAB
+    return detector
```
Comment (KarlPalmen) [inline scripts/SANS/sans/common/general_functions.py]: Some time in the future we should get this info from the IDF or something loaded from it.

**Comment by KarlPalmen** [comment]
Looks good :shipit:

File: scripts/SANS/sans/common/general_functions.py (hunk 1 above)
Comment (AntonPiccardoSelg) [inline scripts/SANS/sans/common/general_functions.py]: I agree completely, this is what I was referring to in the instructions as being not an ideal solution.

Comment (KarlPalmen) [inline scripts/SANS/sans/common/general_functions.py]: I was just being a little more specific about it.

**Pull request merged**